src/main/java/gregtech/api/recipes/ingredients/GTRecipeFluidInput.java

```diff
@@ -67,30 +68,30 @@ public boolean acceptsFluid(@Nullable FluidStack input) {
         return false;
     }
 
+    @Override
+    protected int computeHash() {
+        if (nbtMatcher == null) {
+            //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
+            return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, inputStack.tag);
```

Comment:
What is reason behind different inputs for hash if `nbtMatcher == null`?

Reply:
I've revisited the code, and figured out the original reason behind the different hash strategy.

The reason why fluid NBT gets included only when NBT matcher is not present, is because the behavior of fluid input changes depending on the presence of NBT matcher. If matcher is present, input matching will use NBT matcher for check. If matcher is not present, instead the fluid NBT will be equality checked with ingredient fluid NBT.

Based on this, `equals` implementation has been updated accordingly.